tests/cupyx_tests/scipy_tests/linalg_tests/test_decomp_lu.py

```diff
@@ -32,6 +35,34 @@ def test_lu_factor(self, dtype):
         cupy.testing.assert_allclose(result_cpu[0], result_gpu[0], atol=1e-5)
         cupy.testing.assert_array_equal(result_cpu[1], result_gpu[1])
 
+    @testing.for_float_dtypes(no_float16=True)
+    def test_lu_factor_reconstruction(self, dtype):
+        m, n = self.shape
+        array = cupy.random.randn(m, n, dtype=dtype)
+        lu, piv = cupyx.scipy.linalg.lu_factor(array)
+        # extract ``L`` and ``U`` from ``lu``
+        L = cupy.tril(lu, k=-1)
+        cupy.fill_diagonal(L, 1.)
+        if m < n:
+            L = L[:, :m]
+        U = cupy.triu(lu)
+        if m > n:
+            U = U[:n, :]
+        # apply pivot (on CPU since slaswp is not available in cupy)
+        piv = cupy.asnumpy(piv)
+        rows = numpy.arange(array.shape[0])
+        for i, row in enumerate(piv):
+            if i != row:
+                rows[i], rows[row] = rows[row], rows[i]
+        # revert pivot
+        reversed_piv = numpy.empty_like(rows)
+        reversed_piv[rows] = numpy.arange(rows.size)
```

Comment:
Currently, this test checks A = P^{-1} L U. How about fixing to check PA=LU to avoid reversing `piv`? It will make the test simpler.

Reply:
👍 I edited to simplify the test. I also renamed `array` to `A`. Even if this does not follow pep8, it makes the test closer to the documentation of cuSOLVER, so is more readable IMHO.